mods/tnt/init.lua

```diff
@@ -160,6 +162,43 @@ local function add_effects(pos, radius)
 		maxsize = 16,
 		texture = "tnt_smoke.png",
 	})
+
+	local item, texture = "", "tnt_blast.png" --fallback texture
+	local objs = minetest.get_objects_inside_radius(pos, 2)
+
+	-- we just dropped some items. Look at the items entities and pick
+	-- one of them to use as texture
+	for _, obj in pairs(objs) do
+		if obj and obj:get_luaentity() then
+			item = obj:get_luaentity().itemstring:match("^(.*)%s")
```

Comment:
You can use `ItemStack(itemstring)` here and you get access to `get_name` and `get_definition`.